juju/juju
# Add NetworkInfo interface as a replacement for NetworkConfig

## Description of change
1. NetworkConfig is obsoleted by NetworkInfo
2. network-get now supports 'full' (and not only --primary-address) mode returning more information about the network settings for bindings
3. unit-get private-address is now space aware and returns the address for the default binding for the unit and not 'general' private address as before

## QA steps
For network-get - launch an app with binding X, check that network-get X returns whole set of information (interface name, MAC, IP addresses)
For unit-get private-address:
1. Deploy a machine in two spaces:
juju add-machine --constraints "spaces=space1,space2"
2. Deploy an app to this machine with default binding to space1:
juju deploy foobar --to 0 --bind space1
3. Check that unit-get private-address in this unit context returns address from space1
4. remove app & redeploy with --bind space2, check that unit-get private-address returns address from space2

## Documentation changes
1. network-get has a 'full' mode now
2. unit-get private-address has different behaviour

## Bug reference
This fixes https://bugs.launchpad.net/juju/+bug/1680100
